pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -805,6 +807,22 @@ public void reloadLookUp() throws PulsarClientException {
         return metadataFuture;
     }
 
+    @Override
+    public CompletableFuture<List<String>> getPartitionsForTopic(String topic) {
+        return getPartitionedTopicMetadata(topic).thenApply(metadata -> {
+            if (metadata.partitions > 1) {
+                TopicName topicName = TopicName.get(topic);
+                List<String> partitions = new ArrayList<>(metadata.partitions);
+                for (int i = 0; i < metadata.partitions; i++) {
+                    partitions.add(topicName.getPartition(i).toString());
+                }
+                return partitions;
+            } else {
+                return Collections.singletonList(topic);
```

Comment:
+1 I think failing it with an exception is better

Reply:
If we give exception here, we're forcing an application to handle both cases. 

Eg: I want to create a reader on a topic, I don't know if it's partitioned or not: 

```java
client.getPartitionsForTopic("my-topic")
   .join()
   .forEach(partition -> client.newReader().topic(partition).create());
```

If `getPartitionsForTopic()` returns error, application would always have to differentiate between the 2 cases.

Reply:
ok.. but I think it might not be difficult or complex for client to handle it. exception might be helpful to client if client wants to handle non-partition topic differently. as method name suggests `getPartitionsForTopic(..)` client may expects partitioned-topics in response. 
However, if we still want to return non-partitioned-topic then probably we can add documentation to clarify the behavior.

Reply:
I had added to the javadoc : 

```
  * If the topic is partitioned, this will return a list of partition names. 
  * If the topic is not partitioned, the
  * returned list will contain the topic name itself.
```